src/wallet/wallet.service.ts

```diff
@@ -114,23 +117,33 @@ export class WalletService implements OnModuleInit {
 
   /**
    * Signs a message to deposit buffered ethers
-   * @param prefix - unique prefix from the contract for this type of message
-   * @param depositRoot - current deposit root from the deposit contract
-   * @param keysOpIndex - current index of keys operations from the registry contract
-   * @param blockNumber - current block number
-   * @param blockHash - current block hash
+   * @param signDepositDataParams - parameters for signing deposit message
+   * @param signDepositDataParams.prefix - unique prefix from the contract for this type of message
+   * @param signDepositDataParams.depositRoot - current deposit root from the deposit contract
+   * @param signDepositDataParams.keysOpIndex - current index of keys operations from the registry contract
+   * @param signDepositDataParams.blockNumber - current block number
+   * @param signDepositDataParams.blockHash - current block hash
+   * @param signDepositDataParams.stakingModuleId - target module id
    * @returns signature
    */
-  public async signDepositData(
-    prefix: string,
-    depositRoot: string,
-    keysOpIndex: number,
-    blockNumber: number,
-    blockHash: string,
-  ): Promise<Signature> {
+  public async signDepositData({
+    prefix,
+    blockNumber,
+    blockHash,
+    depositRoot,
+    keysOpIndex,
+    stakingModuleId,
+  }: SignDepositDataParams): Promise<Signature> {
     const encodedData = defaultAbiCoder.encode(
-      ['bytes32', 'bytes32', 'uint256', 'uint256', 'bytes32'],
-      [prefix, depositRoot, keysOpIndex, blockNumber, blockHash],
+      ['bytes32', 'uint256', 'bytes32', 'bytes32', 'uint24', 'uint256'],
+      [
+        prefix,
+        blockNumber,
+        blockHash,
+        depositRoot,
+        stakingModuleId,
+        keysOpIndex,
```

Comment:
stakingModuleId  is `uint256`

https://github.com/lidofinance/lido-dao/blob/feature/shapella-upgrade/contracts/0.8.9/DepositSecurityModule.sol#L406-L413